completion/all.go

```diff
@@ -92,3 +99,35 @@ func generateCompletion(
 		return nil
 	}
 }
+
+func InstallShellCompletion(shell Shell) error {
+	path, err := shell.InstallPath()
+	if err != nil {
+		return fmt.Errorf("get install path: %w", err)
+	}
+
+	err = os.MkdirAll(filepath.Dir(path), 0o755)
+	if err != nil {
+		return fmt.Errorf("create directories: %w", err)
+	}
+
+	if shell.UsesOwnFile() {
+		err := os.WriteFile(path, nil, 0o644)
+		if err != nil {
+			return fmt.Errorf("create file: %w", err)
+		}
+	}
```

Comment:
This "UsesOwnFile" thing is a bit confusing to me. I understand how it works by reading the code but I think it hurts the API. For example, this function is idempotent only for particular shells. That kind of divergent behavior causes bugs.

I wonder if we could remove that concept entirely and instead use markers a la `coder config-ssh`. E.g. `# BEGIN COMPLETION ...` in the path.

---

Also, in the "UsesOwnFile" world, this code would be more cleanly expressed by conditionally using `os.O_TRUNC` on the `OpenFile`. It's more readable, performant, and concise.